drf_attachments/admin.py

```diff
@@ -117,6 +117,65 @@ def download_view(self, request, object_id):
         return response
 
 
+class DynamicallyDisabledAttachmentInlineForm(AttachmentForm):
+    class Meta:
+        model = Attachment
+        fields = "__all__"
+
+    def disable_inline_fields(self):
+        """
+        Override to return True/False when custom condition is met
+        and change field readonly status in inline admin list on True
+        """
+        raise NotImplementedError()
+
+    def __init__(
+        self,
+        data=None,
+        files=None,
+        auto_id="id_%s",
+        prefix=None,
+        initial=None,
+        error_class=ErrorList,
+        label_suffix=None,
+        empty_permitted=False,
+        instance=None,
+        use_required_attribute=None,
+        renderer=None,
+    ):
+        super().__init__(
+            data,
+            files,
+            auto_id,
+            prefix,
+            initial,
+            error_class,
+            label_suffix,
+            empty_permitted,
+            instance,
+            use_required_attribute,
+            renderer,
+        )
+
+        # make all fields of the inline attachment read only
+        if self.disable_inline_fields():
+            for field in self.fields:
+                self.fields[field].disabled = True
+
+
+class DynamicallyDisabledAttachmentInlineFormSet(BaseGenericInlineFormSet):
+    def add_fields(self, form, index):
+        """
+        Override to disable the DELETE checkbox of entries matching a specific custom condition.
+        E.g. to make this dependent of the DynamicallyDisabledAttachmentInlineForm's disable_inline_fields method:
+
+        super().add_fields(form, index)
+        if hasattr(form, 'disable_inline_fields') and form.disable_inline_fields():
+            form.fields['DELETE'].disabled = True
+        """
+        raise NotImplementedError()
```

Comment:
If we inherit from this and use `super().add_fields(form, index)` we get a `NotImplementedError` because the method raises it. I suggest using the example implementation as the actual implementation here:

```suggestion
class DynamicallyDisabledAttachmentInlineFormSet(BaseGenericInlineFormSet):
    def add_fields(self, form, index):
        """
        Disables the DELETE checkbox of entries matching a specific custom condition.
        """
        super().add_fields(form, index)
        if hasattr(form, 'disable_inline_fields') and form.disable_inline_fields():
            form.fields['DELETE'].disabled = True
```

Reply:
Done, but I changed the description text to `Disables the DELETE checkbox of disabled inline entries.`.